### PR TITLE
Sx21 search room

### PR DESCRIPTION
**Pull request by ahmad-diaa**
#22

@daliamaarek , @Youmna-Salah , @maggie-moheb  , @AmirMaurice , @ahmad3m3   please review :)

@AmirMaurice : UML
@daliamaarek  : Documentation
@Youmna-Salah : Code
@maggie-moheb : Test
@ahmad3m3 : Scenario

### CODE REVIEW

**Comment by ahmad-diaa** [comment]
@Youmna-Salah changed, please review again

**Comment by AmirMaurice** [comment]
@ahmad-diaa 
concerning the sequence diagram you have to add an entity for rails Controller where method search exists

**Comment by ahmad-diaa** [comment]
@AmirMaurice  the search is done within Android listview only, rails search method is not used

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/CustomListAdapter.java [hunk 1]
```diff
@@ -9,29 +10,80 @@
...
+import java.util.Locale;
+
 public class CustomListAdapter extends ArrayAdapter<String> {
 
-    private final Activity context;
-    private final String[] itemname;
-    private final Integer[] imgid;
+    private Activity context;
+    private ArrayList<String> itemName;
+
```
Comment (Youmna-Salah) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/CustomListAdapter.java]: blank space :(

File: SmartX/app/src/main/java/com/smartx/cookies/smartx/CustomListAdapter.java [hunk 2]
```diff
@@ -9,29 +10,80 @@
...
+    public ArrayList<Integer> getImgId() {
+        return imgId;
+    }
+
+    public void setImgId(ArrayList<Integer> imgId) {
+        this.imgId = imgId;
+    }
+
+    private ArrayList<Integer> imgId;
+
```
Comment (Youmna-Salah) [inline SmartX/app/src/main/java/com/smartx/cookies/smartx/CustomListAdapter.java]: blank line

**Comment by maggie-moheb** [comment]
The test is running perfectly

**Comment by ahmad-diaa** [comment]
@Youmna-Salah fixed

**Comment by AmirMaurice** [comment]
@ahmad-diaa perfect you are right :+1:

**Comment by ahmad3m3** [comment]
@ahmad-diaa : It works fine :+1:

**Comment by daliamaarek** [comment]
@ahmad-diaa Change of plans, getters and setters should have documentation too! :hamburger:

**Comment by ahmad-diaa** [comment]
@daliamaarek  done

**Pull request merged**